web-api/workflow-terraform/migration/main/lambdas/migrations/0013-default-case-status-history.ts

```diff
@@ -0,0 +1,18 @@
+import { TDynamoRecord } from '../../../../../src/persistence/dynamo/dynamoTypes';
+
+const isCaseRecord = item => {
+  return item.pk.startsWith('case|') && item.sk.startsWith('case|');
+};
+
+export const migrateItems = items => {
+  const itemsAfter: TDynamoRecord[] = [];
+
+  for (const item of items) {
+    if (isCaseRecord(item) && !item.caseStatusHistory) {
+      item.caseStatusHistory = [];
+    }
+    itemsAfter.push(item);
+  }
+
```

Comment:
Why do you think we needed to write a migration for this? 

Case.ts already has

```
    this.caseStatusHistory = rawCase.caseStatusHistory || [];
```

which treats it as an empty array if it doesn't have there.

Reply:
Good question! The Case constructor does default that value to an empty array, but there are places where we return a case _record_, straight from DynamoDB, that has not been passed through the case constructor. We felt this was the best decision so that the joi validation and the definition of case match, regardless of whether or not an item has been passed through the Case constructor.